pkg/auth/auth.go

```diff
@@ -34,8 +35,20 @@ func CheckAuthFile(authfile string) error {
 }
 
 // Login login to the server with creds from Stdin or CLI
-func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginOptions, registry string) error {
-	server := getRegistryName(registry)
+func Login(ctx context.Context, systemContext *types.SystemContext, opts *LoginOptions, registry []string) error {
+	var server string
+	if len(registry) == 0 {
+		registriesFromFile, err := registries.GetRegistries()
```

Comment:
Just call `sysregistriesv2.UnqualifiedSearchRegistries` _directly_ here; notably with the caller-specified `systemContext` instead of ignoring that and building an entirely new one.